src/blueprints/tv2/tv2-end-state-for-part-calculator.ts

```diff
@@ -60,20 +60,30 @@ export class Tv2EndStateForPartCalculator implements BlueprintGetEndStateForPart
     return endState
   }
 
+  private getEmptyTv2RundownPersistentState(): Tv2RundownPersistentState {
+    return {
+      activeMediaPlayerSessions: [],
+      isNewSegment: false
+    }
+  }
+
   private calculateSisyfosPersistenceMetaData(
     part: Part,
     previousPartEndState: Tv2PartEndState | undefined,
     time: number,
     rundownPersistentState: Tv2RundownPersistentState
-  ): Tv2SisyfosPersistenceMetaData {
-    const layersWantingToPersist: string[] =
-        !rundownPersistentState?.isNewSegment &&
-        previousPartEndState &&
-        previousPartEndState.sisyfosPersistenceMetaData
-          ? previousPartEndState.sisyfosPersistenceMetaData.sisyfosLayers
-          : []
+  ): Tv2SisyfosPersistenceMetadata {
+    const layersWantingToPersist: string[] = this.findLayersWantingToPersist(rundownPersistentState, previousPartEndState)
     return {
       sisyfosLayers: this.sisyfosPersistentLayerFinder.findLayersToPersist(part, time, layersWantingToPersist),
     }
   }
+
+  private findLayersWantingToPersist(rundownPersistentState: Tv2RundownPersistentState, previousPartEndState: Tv2PartEndState | undefined): string[] {
+    return !rundownPersistentState.isNewSegment &&
+    previousPartEndState &&
+    previousPartEndState.sisyfosPersistenceMetadata
+      ? previousPartEndState.sisyfosPersistenceMetadata.sisyfosLayers
+      : []
```

Comment:
The indentation is a bit funny here. Alternatively, it can be changed to a guard. And by using optional chaining, you can remove `&& previousPartEndState`. Note: In the snippet below the condition is negated.
```typescript
    if (rundownPersistentState.isNewSegment || !previousPartEndState?.sisyfosPersistenceMetadata) {
      return []
    }
    return previousPartEndState.sisyfosPersistenceMetadata.sisyfosLayers
```

Reply:
Updated.